java/src/main/java/com/mlt/converter/triangulation/TriangulationUtils.java

```diff
@@ -0,0 +1,76 @@
+package com.mlt.converter.triangulation;
+
+import earcut4j.Earcut;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.ArrayUtils;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.MultiPolygon;
+import org.locationtech.jts.geom.Polygon;
+
+public class TriangulationUtils {
+  private TriangulationUtils() {}
+
+  public static TriangulatedPolygon triangulatePolygon(Polygon polygon) {
+    var convertedCoordinates = convertCoordinates(polygon.getCoordinates());
+
+    List<Integer> triangles = Earcut.earcut(convertedCoordinates, null, 2);
+
+    ArrayList<Integer> indexBuffer = new ArrayList<>(triangles);
+    var numTriangles = triangles.size() / 3;
+
+    return new TriangulatedPolygon(indexBuffer, numTriangles);
+  }
+
+  public static TriangulatedPolygon triangulatePolygonWithHoles(MultiPolygon multiPolygon) {
+    var holeIndex = 0;
+
+    ArrayList<Double> multiPolygonCoordinates = new ArrayList<>();
+    ArrayList<Integer> holeIndices = new ArrayList<>();
+
+    for (int i = 0; i < multiPolygon.getNumGeometries(); i++) {
+      // assertion: first polygon defines the outer linear ring and the other polygons define its
+      // holes!
+      if (i == 0) {
+        holeIndex = multiPolygon.getGeometryN(i).getCoordinates().length;
```

Comment:
Each call to getCoordinates may be transforming a packed double[] CoordinateSequence to a `Coordinate[]` array. Is it possible to just extract the `CoordinateSequence` once and use `double getX(int idx)`, `double getY(int idx)` and `double getZ(int idx)` (only when coordinateSequence.getDimension() > 2)